packages/compiler-sfc/src/script/resolveType.ts

```diff
@@ -764,6 +766,21 @@ function innerResolveTypeReference(
           !ns.declare,
         )
       }
+    } else {
+      let importType: Import
+      if (
+        (importType = scope.imports[name[0]]) &&
+        importType.imported === '*'
+      ) {
```

Comment:
Is there a reason to have the assignment inside the `if` condition? Wouldn't it be simpler to do the assignment straight away?

```ts
const importType = scope.imports[name[0]]
```

I also wonder whether the `if` check could use optional chaining (which is allowed in this file):

```ts
if (importType?.imported === '*') {
```

Reply:
Updated~